### DISCUSSION TITLE
MinZoom and MaxZoom?

### DISCUSSION THREAD

**Discussion by FrozDark** [Mapsui/Mapsui#1315] [Q&A] [answered]
Is there any way to limit the zoom?

**Comment by pauldendulk** [comment, marked as answer]
You can use MapControl.Map.Limiter.ZoomLimits for that. 

In master there is a 'Keep Center in Map' sample under 'Navigation'. It uses the resolution values. If you want to know the indexes that correspond to those values look at MapControl.Map.Resolutions.

**Reply by janusw** [reply]
What units does `MapControl.Map.Limiter.ZoomLimits` have? The `KeepCenterInMapSample` uses:

```
ZoomLimits = new MinMax(0.15, 2500);
```

What does that mean? You say it uses "resolution values", but how are those defined? It this what `ZoomLevelExtensions.ToMapsuiResolution` computes? It seems to me such things should be better documented ...

**Reply by pauldendulk** [reply]
Good point, in fact when I posted this answer I was afraid FrozDark would ask for a clarification of the resolution, but I got away with it 😁.

I just wrote a documentation page here:
https://github.com/Mapsui/Mapsui/blob/master/docfx/documentation/resolution.md

Unfortunately it may still not be very easy to understand for a user. The short answer is:
> The resolution of a viewport is its size in map coordinates (of the coordinate system used) divided by its size in pixels.

or
> var resolution = viewport.Extent.Width / viewport.Width;

but does that answer your questions?

**Reply by pauldendulk** [reply]
@janusw If you have any feedback on the documentation please let me know. I found it quite hard to write. I have to keep different kind of users in mind and it is easy to go into too much detail https://github.com/Mapsui/Mapsui/blob/master/docfx/documentation/resolution.md.

**Reply by janusw** [reply]
Thanks, Paul. Your new documentation page helps a lot. Maybe it would make sense to link to this from the relevant places in the API docu? For example:
* https://mapsui.com/api/Mapsui.UI.IViewportLimiter.html#Mapsui_UI_IViewportLimiter_ZoomLimits
* https://mapsui.com/api/Mapsui.IMap.html#Mapsui_IMap_Resolutions
* etc
